src/lib/redux/inventoryData.ts

```diff
@@ -0,0 +1,39 @@
+import { InventoryRecord, InventoryUpdateRecord, PurchaseRequestRecord, SiteRecord } from '../airtable/interface';
+import { getAllProducts, getInventorysByIds, getInventoryUpdatesByIds, getPurchaseRequestsByIds } from '../airtable/request';
+import { saveInventoryData, setLoadingForInventoryData } from './inventoryDataSlice';
+import { store } from './store';
+
+const refreshInventoryData = async (currentSite: SiteRecord): Promise<void> => {
+  store.dispatch(setLoadingForInventoryData());
+  if (currentSite) { // TODO @wangannie what to do if site null
+    const products = await getAllProducts();
+
+    const currentSiteInventory = await getInventorysByIds(currentSite.inventoryIds);
+    
+    // TODO @wangannie: look into avoiding .flat, simplifying/eliminating .filter
+    const inventoryWithPurchaseRequests = currentSiteInventory.filter((inventory: InventoryRecord) => inventory.purchaseRequestIds);
+    let purchaseRequests: PurchaseRequestRecord[] = [];
+    await Promise.all(inventoryWithPurchaseRequests.map(async (inventory: InventoryRecord) => 
+      getPurchaseRequestsByIds(inventory.purchaseRequestIds)
```

Comment:
We generally wanna minimize API requests whenever possible. Wouldn't it be faster to gather all the `purchaseRequestIds` for all inventory, make 1 call to `getPurchaseRequestsByIds` and then do matching of `purchaseRequests` to their inventory afterwards?